tests/cpp/test_multidevice_overlap.cpp

```diff
@@ -332,7 +374,151 @@ TEST_F(OverlapTest, ReduceScatterBasedPipeliningHostIrImplementation) {
 
     hie.runWithInput(std::move(inputs));
   }
-  validate();
 }
 
+using RingBasedOverlapTestParams = std::tuple<
+    bool, // use_coalescence
```

Comment:
```suggestion
    // Whether to coalesce P2P communication. ProcessGroupNCCL provides two methods startCoalescing and endCoalescing, which internally correspond to ncclGroupStart and ncclGroupEnd, see [doc here](https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/usage/groups.html). Those calls group p2p calls that need to be progressed together -- one global work handle returned by endCoalescing needs to be progressed. It is in general preferable to coalesce send/recv calls. The only drawback is that we don't have a fine-grain control on synchronicity, in other words, we can only synchronize with the bulked communication as a whole. ProcessGroupUCC does not implement coalesced groups for now. It should not be a problem to achieve full bidirectional BW with two send/recv though. However, having more than two ops in a batch will be suboptimal. Adding UCC coalescing was discussed but not added to POR for now because lacking a good use case.
    bool,
```